exp-player/addon/components/exp-lookit-dialogue-page/component.js

```diff
@@ -0,0 +1,797 @@
+            "kind": "exp-lookit-dialogue-page",
+            "id": "phase-5",
+            "baseDir": "https://s3.amazonaws.com/lookitcontents/politeness/",
+            "audioTypes": ["mp3", "ogg"],
+            "backgroundImage": "order1_test1_background.png",
+            "doRecording": false,
+            "autoProceed": false,
+            "isChoiceFrame": true,
+            "parentTextBlock": {
+                "title": "Parents:",
+                "text": "click on the character your child selects.",
+                "emph": true
+            },
+            "images": [
+                {
+                    "id": "speaker1",
+                    "src": "order1_test1_speaker1.png",
+                    "left": "20",
+                    "bottom": "2",
+                    "height": "60"
+                },
+                {
+                    "id": "speaker2",
+                    "src": "order1_test1_speaker2.png",
+                    "left": "60",
+                    "bottom": "2",
+                    "height": "60"
+                }
+            ],
+            "audioSources": [
+                {
+                    "audioId": "firstAudio",
+                    "sources": [
+                        {
+                            "stub": "polcon_example_5q1"
+                        }
+                    ]
+                }
+            ]
+        }
+ }
+
+ * ```
+ * @class ExpLookitDialoguePage
+ * @extends ExpFrameBaseUnsafe
+ * @uses FullScreen
+ * @uses VideoRecord
+ */
+
+export default ExpFrameBaseUnsafeComponent.extend(FullScreen, VideoRecord,  {
+    // In the Lookit use case, the frame BEFORE the one that goes fullscreen
+    // must use "unsafe" saves (in order for the fullscreen event to register as
+    // being user-initiated and not from a promise handler) #LEI-369.
+    // exp-alternation frames are expected to be repeated, so they need to be
+    // unsafe.
+    type: 'exp-lookit-dialogue-page',
+    layout: layout,
+    displayFullscreen: true, // force fullscreen for all uses of this component
+    fullScreenElementId: 'experiment-player',
+    fsButtonID: 'fsButton',
+    videoRecorder: Ember.inject.service(),
+    recorder: null,
+    hasCamAccess: Ember.computed.alias('recorder.hasCamAccess'),
+    videoUploadConnected: Ember.computed.alias('recorder.connected'),
+
+    // Track state of experiment
+    completedAudio: false, // for main narration audio
+    imageAudioCompleted: new Set(),
+    currentlyHighlighted: null, // id for image currently selected
+
+    currentAudioIndex: -1, // during initial sequential audio, holds an index into audioSources
+
+    // Can the user click the 'next' button yet? Require all 'main' audio to
+    // have played. For a choice frame, require that one of the images is
+    // selected; for other frames, require that any required image-audio has
+    // completed.
+    readyToProceed: Ember.computed('completedAudio', 'imageAudioCompleted', 'currentlyHighlighted',
```

Comment:
Small note: `imageAudioCompleted` is a `Set()`. I like them in theory, but they may not play nice with ember. Two things to watch for:

1. Computed properties do not automatically know to watch the *contents* of an array unless you [use](https://guides.emberjs.com/v2.6.0/object-model/computed-properties-and-aggregate-data/) `arrayName.[]` or similar.   (for maintainability, this latter syntax is highly recommended here)

2. Off the top of my head I don't recall whether ember knows how to watch sets (unlike arrays, which ember wraps in magic). If you don't need fast lookups on a lot of data, the potential downsides for maintainability (needing to unexpectedly use `notifyPropertyChange`) might outweigh the benefits by creating room for potential bugs. 

This won't block merge for now, but watch this closely for the future; you might simplify your code (and reduce bug potential) by using a regular array + `arrayName.[]`.